owr/owr_bus.h

```diff
@@ -0,0 +1,139 @@
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ */
+
+/*/
+\*\ OwrBus
+/*/
+
+#ifndef __OWR_BUS_H__
+#define __OWR_BUS_H__
+
+#include "owr_message_origin.h"
+#include "owr_types.h"
+
+#include <glib-object.h>
+
+G_BEGIN_DECLS
+
+typedef enum {
+    OWR_MESSAGE_TYPE_ERROR = (1 << 0),
+    OWR_MESSAGE_TYPE_STATS = (1 << 1),
+    OWR_MESSAGE_TYPE_EVENT = (1 << 2)
+} OwrMessageType;
+
+/**
+ * OwrMessageSubType:
+ *
+ * @OWR_ERROR_TYPE_GST_ERROR: a GStreamer error message was received
+ * - @debug: #gchar* a string containing debug information
+ * - @code: #gint the error code
+ * - @domain: #gchar the error domain
+ * - @message: #gchar the error message
+ *
+ * @OWR_STATS_TYPE_SCHEDULE: information about a call that was scheduled on the main thread
+ * - @start_time: #gint64 monotonic time when the call was scheduled
+ * - @call_time: #gint64 monotonic time when the function call started
+ * - @end_time: #gint64 monotonic time when the function call completed
+ *
+ * @OWR_STATS_TYPE_SEND_PIPELINE_ADDED: a send pipeline was added to a media session
+ * - @start_time: #gint64 monotonic time when the pipeline setup began
+ * - @end_time: #gint64 monotonic time when the pipeline setup was completed
+ *
+ * @OWR_STATS_TYPE_SEND_PIPELINE_REMOVED: a send pipeline was removed to a media session
+ * - @start_time: #gint64 monotonic time when the pipeline teardown began
+ * - @end_time: #gint64 monotonic time when the pipeline teardown was completed
+ *
+ * @OWR_EVENT_TYPE_RENDERER_STARTED: a renderer was started
+ *
+ * @OWR_EVENT_TYPE_RENDERER_STOPPED: a renderer was stopped
+ *
+ * @OWR_EVENT_TYPE_LOCAL_SOURCE_STARTED: a local media source was started
+ * - @start_time: #gint64 monotonic time when the pipeline setup began
+ * - @end_time: #gint64 monotonic time when the pipeline setup was completed
+ *
+ * @OWR_EVENT_TYPE_LOCAL_SOURCE_STOPPED: a local media source was stopped
+ * - @start_time: #gint64 monotonic time when the pipeline teardown began
+ * - @end_time: #gint64 monotonic time when the pipeline teardown was completed
+ *
+ * @OWR_EVENT_TYPE_GST_EOS: an GStreamer End-of-stream message was received
+ */
+typedef enum {
+    OWR_ERROR_TYPE_TEST = 0x1000,
+    OWR_ERROR_TYPE_GST_ERROR,
+    OWR_STATS_TYPE_TEST = 0x2000,
+    OWR_STATS_TYPE_SCHEDULE,
+    OWR_STATS_TYPE_SEND_PIPELINE_ADDED,
+    OWR_STATS_TYPE_SEND_PIPELINE_REMOVED,
+    OWR_EVENT_TYPE_TEST = 0x3000,
+    OWR_EVENT_TYPE_GST_EOS,
```

Comment:
Again, that it is a GStreamer EOS is not interesting to the application, just that some kind of end of stream occurred and a bit more text about what that entails would be good.